### PR TITLE
#162725250 Setup and deploy base application to heroku

### PR DESCRIPTION
**Pull request by armstrongsouljah**
#### What does this PR do?
This PR is to set up and deploy base application to Heroku.

#### Description of the task to be completed.
Setup and deploy base application to Heroku so that other users can access the application.

#### how should this be manually tested?
- First things first, fetch this branch by running `git fetch origin ch-setup-app-and-deploy-to-heroku-162725250`
- Check into this branch by running ` git checkout -- ch-setup-app-and-deploy-to-heroku-162725250`
- Activate your virtual environment by running `source venv/bin/activate`
- Install dependencies by running `pip install -r requirements.txt`
-  View  live `staging app`  [here](https://ah-backend-staging.herokuapp.com/api/)

To run locally, you need to have a `heroku-cli` installed on your machine
   [Link on how to install heroku cli](https://devcenter.heroku.com/articles/heroku-cli)
    - run local Heroku  local server using `heroku local web`
   - navigate to `localhost:5000/api/` in your web browser

#### Relevant stories
[#162725250](https://www.pivotaltracker.com/story/show/162725250)

### CODE REVIEW

File: authors/urls.py [hunk 1]
```diff
@@ -22,3 +24,6 @@
     path('api/', include(('authors.apps.authentication.urls', 'authentication'), namespace='authentication')),
 ]
 
+if settings.DEBUG:
```
Comment (MuhweziDeo) [inline authors/urls.py:28]: @armstrongsouljah Are u collecting static files locally

File: authors/settings.py [hunk 2]
```diff
@@ -19,12 +20,16 @@
 # See https://docs.djangoproject.com/en/1.11/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = '7pgozr2jn7zs_o%i8id6=rddie!*0f0qy3$oy$(8231i^4*@u3'
+SECRET_KEY = config('SECRET_KEY')
 
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = True
```
Comment (malfahad) [inline authors/settings.py]: DEBUG should not be true in production.
 Can you consider adding an IS_HEROKU check and set DEBUG accordingly?

**Review by malfahad — CHANGES_REQUESTED** [review]

**Comment by armstrongsouljah** [comment]
Thanks for pointing out the `DEBUG` let me change it accordingly.

File: authors/settings.py (hunk 2 above)
Comment (armstrongsouljah) [reply authors/settings.py]: @malfahad I am working on editing `DEBUG` to use the environment variables on both the staging and production environments. Thus i won't need  `IS_HEROKU`, am instead going to use `IS_STAGING` and `IS_PRODUCTION`.

Comment (armstrongsouljah) [reply authors/settings.py]: @malfahad and team, I have updated the `DEBUG` constant in `settings.py` to point to environment variables on Heroku for both staging and production environments.

**Review by malfahad — DISMISSED** [review]

**Review by calvinpete — DISMISSED** [review]
@armstrongsouljah Everything looks fine to me

**Comment by mariamiah** [comment]
@armstrongsouljah Code looks okay but can you grant access or add me as a collaborator, seems I cannot access the application

**Comment by armstrongsouljah** [comment]
@mariamiah. I used  Deo Muhwezi's account to create the applications and
he's the only one with access to the application.  Only he can add you as a
collaborator.

**Review by MuhweziDeo — DISMISSED** [review]

**Review by malfahad — DISMISSED** [review]

**Review by MuhweziDeo — DISMISSED** [review]

**Review by d-rita — DISMISSED** [review]

**Review by MuhweziDeo — DISMISSED** [review]

**Review by kdaglas — DISMISSED** [review]
Makes more sense after your explanation, so great stuff

**Review by malfahad — DISMISSED** [review]

**Review by calvinpete — DISMISSED** [review]
@armstrongsouljah LGTM

**Review by mariamiah — DISMISSED** [review]
LGTM

**Review by d-rita — DISMISSED** [review]

**Review by MuhweziDeo — APPROVED** [review]

**Review by calvinpete — APPROVED** [review]
@armstrongsouljah LGTM

**Review by mariamiah — APPROVED** [review]

**Pull request merged**